init-tools.sh

```diff
@@ -12,106 +12,103 @@ __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
+__BUILD_TOOLS_SEMAPHORE=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION/init-tools.complete
 
-if [ -z "$__DOTNET_PKG" ]; then
-    if [ "$(uname -m | grep "i[3456]86")" = "i686" ]; then
-        echo "Warning: build not supported on 32 bit Unix"
-    fi
-OSName=$(uname -s)
-    case $OSName in
-        Darwin)
-            OS=OSX
-            __DOTNET_PKG=dotnet-dev-osx-x64
-            ulimit -n 2048
-            ;;
-
-        Linux)
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            OS=Linux
-            ;;
-
-        *)
-            echo "Unsupported OS '$OSName' detected. Downloading linux-x64 tools."
-            OS=Linux
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            ;;
-  esac
+if [ -e $__BUILD_TOOLS_SEMAPHORE ]; then
+    echo "Tools are already initialized"
+    exit 0
 fi
 
-if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; then
-    __PATCH_CLI_NUGET_FRAMEWORKS=0
-
-    if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
-    echo "Running: $__scriptpath/init-tools.sh" > $__init_tools_log
-
-    if [ ! -e $__DOTNET_PATH ]; then
-
-        mkdir -p "$__DOTNET_PATH"
-
-        if [ -n "$DOTNET_TOOLSET_DIR" ] && [ -d "$DOTNET_TOOLSET_DIR/$__DOTNET_TOOLS_VERSION" ]; then
-            echo "Copying $DOTNET_TOOLSET_DIR/$__DOTNET_TOOLS_VERSION to $__DOTNET_PATH" >> $__init_tools_log
-            cp -r $DOTNET_TOOLSET_DIR/$__DOTNET_TOOLS_VERSION/* $__DOTNET_PATH
-        elif [ -n "$DOTNET_TOOL_DIR" ] && [ -d "$DOTNET_TOOL_DIR" ]; then
-            echo "Copying $DOTNET_TOOL_DIR to $__DOTNET_PATH" >> $__init_tools_log
-            cp -r $DOTNET_TOOL_DIR/* $__DOTNET_PATH
-        else
-            echo "Installing dotnet cli..."
-            __DOTNET_LOCATION="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz"
-            # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-            echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
-            which curl > /dev/null 2> /dev/null
-            if [ $? -ne 0 ]; then
-                wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-            else
-                curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-            fi
-            cd $__DOTNET_PATH
-            tar -xf $__DOTNET_PATH/dotnet.tar
-
-            cd $__scriptpath
-
-            __PATCH_CLI_NUGET_FRAMEWORKS=1
-        fi
+if [ -d "$DotNetBuildToolsDir" ]; then
+    echo Using tools from '$DotNetBuildToolsDir'.
+    ln -s "$DotNetBuildToolsDir" "$__TOOLRUNTIME_DIR"
+
+    if [ -e "$__DOTNET_CMD" ]; then
+        echo ERROR: Could not find dotnet in the overridden toolset at '$__DOTNET_CMD'
+        exit 1
     fi
+    echo Done initializing tools.
+    touch $__BUILD_TOOLS_SEMAPHORE
+    exit 0
+fi
 
+__PATCH_CLI_NUGET_FRAMEWORKS=0
```

Comment:
I must be missing something, but where is this variable used?  I only see it being set.

Reply:
I couldn't find its usage either but I didn't want to remove it as part of this change.

Reply:
I did a search and can't find it either.  I think if we are going to update everyone's init-tools, doing some minimal clean up would be a decent idea.

Reply:
https://github.com/dotnet/corefx/commit/9ff2ca74cbb2d1a7893af3d371d9f003eabf4d51#diff-c02be70df3e835d946f675bd0bedc234R155  this change added it.

https://github.com/dotnet/corefx/commit/8b471f76eef041f0cf40fec65c99a29b235bbe7d#diff-c02be70df3e835d946f675bd0bedc234L154  this change took the usage out without removing the variable.

Reply:
Yep I will clean it out.